### PR TITLE
fuzz: fix inbound fuzz logic expecting wrong IP address

### PR DESCRIPTION
**Pull request by hawkw**
PR #1179 changed the inbound HTTP fuzz logic, and added a mocked
`Target` type similar to other stack tests. However, this introduced a
bug, because the `Target::addr()` function, which returns the socket
address that the mock TCP connector should expect the proxy to try to
connect to, returns the (mocked) original destination address. Since the
stack we build for the fuzz test maps the target IP address to localhost
before calling into the connect stack (as the real proxy currently
does), this means that we attempt to connect to an address the mock
connector doesn't expect. Then, it panics and the fuzz test fails.

This branch fixes this issue by changing `Target::addr` to return
127.0.0.1:80 instead of 192.0.2.2:80. This should match what the proxy
will actually attempt to connect to.

I also updated the fuzz logic to use exactly the same connect stack as
the HTTP stack tests (including a timeout). Although this shouldn't be
exercised, it makes the fuzz test more closely match the stack tests and
thus the real life proxy. I wonder if we can eventually share more code
between the tests and the fuzzers, but that seems like a follow-up.

Signed-off-by: Eliza Weisman <eliza@buoyant.io>

### CODE REVIEW

**Comment by hawkw** [comment]
Also, another thought: Currently, we only try to _build_ the fuzz tests on CI, and never actually run them. This means that the CI build ensures they compile, but doesn't do anything to catch even simple logic errors in the fuzz logic itself. I wonder if we want to check in known-good minimized reproducers for each fuzz test, and run _those_ on CI. That way, the CI build doesn't spend a bunch of time actually running the fuzzer, but we would actually run the fuzz logic once as a sort of smoke test that we didn't just completely break it.

Just a thought.

File: linkerd/app/inbound/src/http/mod.rs [hunk 1]
```diff
@@ -124,7 +124,11 @@ pub mod fuzz {
         I: io::AsyncRead + io::AsyncWrite + io::PeerAddr + Send + Unpin + 'static,
     {
         let connect = svc::stack(connect)
-            .push_map_target(|t: Http| Remote(ServerAddr(([127, 0, 0, 1], t.param()).into())))
+            .map_stack(|cfg, _, s| {
+                s.push_map_target(|p| Remote(ServerAddr(([127, 0, 0, 1], p).into())))
+                    .push_map_target(|t| Param::<u16>::param(&t))
+                    .push_connect_timeout(cfg.proxy.connect.timeout)
+            })
```
Comment (olix0r) [inline linkerd/app/inbound/src/http/mod.rs:131]: is this change necessary? it seems to do the same thing afaict

**Review by olix0r — APPROVED** [review]

File: linkerd/app/inbound/src/http/mod.rs (hunk 1 above)
Comment (hawkw) [reply linkerd/app/inbound/src/http/mod.rs:131]: it's not necessary, the intent was just to ensure this code was in sync with the stack tests. i can back out this change.
